packages/txservice/src/provider.ts

```diff
@@ -461,128 +492,197 @@ export class ChainRpcProvider {
+   */
+  private checkSigner() {
+    if (!this.signer) {
+      throw new NxtpError("Method requires signer, and no signer was provided.");
+    }
+  }
+
+  /**
+   * The RPC method execute wrapper is used for wrapping and parsing errors, as well as ensuring that
+   * providers are ready before any call is made. Also used for executing multiple retries for RPC
+   * requests to providers. This is to circumvent any issues related to unreliable internet/network
+   * issues, whether locally, or externally (for the provider's network).
    *
-   * @returns A ResultAsync instance containing an object of the specified type or an NxtpError.
+   * @param method - The method callback to execute and wrap in retries.
+   * @returns The object of the specified generic type.
+   * @throws NxtpError if the method fails to execute.
    */
-  private resultWrapper<T>(
-    needsSigner: boolean,
-    method: () => Promise<T>,
-    rpcCanTimeout = true,
-  ): ResultAsync<T, NxtpError> {
-    const RPC_TIMEOUT = 60_000;
-    return ResultAsync.fromPromise(
-      this.isReady().then(async () => {
-        if (needsSigner && this.isReadOnly) {
-          throw new NxtpError("Method requires signer, and no signer was provided.");
-        }
-        const errors = [];
-        let result: T | undefined = undefined;
-        // eslint-disable-next-line @typescript-eslint/no-unused-vars
-        for (const _ of Array(5).fill(0)) {
-          try {
-            const _result = await Promise.race(
-              [method()].concat(
-                rpcCanTimeout
-                  ? [
-                      new Promise(async (_res, reject) => {
-                        await delay(RPC_TIMEOUT);
-                        reject(new RpcError(RpcError.reasons.Timeout));
-                      }),
-                    ]
-                  : [],
-              ),
-            );
-            result = _result;
-            break;
-          } catch (e) {
-            const error = parseError(e);
-            // NOTE: If it's an rpc timeout error, we want to go ahead and throw.
-            if (error.type === RpcError.type && (error as RpcError).reason != RpcError.reasons.Timeout) {
-              errors.push(error);
-            } else {
-              throw error;
-            }
-          }
-        }
-        if (result === undefined) {
-          throw new RpcError(RpcError.reasons.FailedToSend, { errors });
+  private async execute<T>(needsSigner: boolean, method: (provider: SyncProvider) => Promise<T>): Promise<T> {
+    // If we need a signer, check to ensure we have one.
+    if (needsSigner) {
+      this.checkSigner();
+    }
+    if (!this.blockPeriod) {
+      // This block should only be called once.
+      const currentBlock = await this.getBlock("latest");
+      const previousBlock = await this.getBlock(currentBlock.parentHash);
+      this.blockPeriod = currentBlock.timestamp - previousBlock.timestamp;
+    }
```

Comment:
nice, so presumably this is the cache period?

Reply:
yeah could be. I can add a TODO for this

Reply:
also thinking that putting this setup in execute is tacky.

Going to move this to a sep method and call it in constructor (without await obvi)... and just leave the default as 1 sec until it gets set.